pulsar/backoff/backoff.go

```diff
@@ -26,10 +26,14 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-// BackoffPolicy parameterize the following options in the reconnection logic to
+// Policy parameterize the following options in the reconnection logic to
 // allow users to customize the reconnection logic (minBackoff, maxBackoff and jitterPercentage)
-type BackoffPolicy interface {
+type Policy interface {
+	// Next returns the delay to wait before next retry
 	Next() time.Duration
+
+	// IsMaxBackoffReached evaluates if the max number of retries is reached
+	IsMaxBackoffReached(delayReconnectTime, totalDelayReconnectTime time.Duration) bool
```

Comment:
`delayReconnectTime` and `totalDelayReconnectTime` are unnecessary, we can record this time in the backoff. 

Usually, we need to reset this backoff when some logic works fine, could you add the `Reset()` to this interface.

Reply:
If you agree with this idea, you can move the backoff object to the global scope from the local scope, which can reduce the allocation of objects.

Reply:
Done.